File: ProjectEuler2.py
# ProjectEuler.net problem 2

def fib_sum_of_even_num(val):
    fib = [1, 2]
    i=1
    sum = 0
    while(fib[i]<=val):
        fib.append(fib[i]+fib[i-1])
        if fib[i]%2==0:
            sum = sum + fib[i]
        i = i+1
    print(sum)



fib_sum_of_even_num(40000000000000000000000000000000000000000)

File: ProjectEuler2.rb
# ProjectEuler.net Problem 2 in Ruby


def sum_of_even_fibonachi(val)
  sequence = [1, 2]
  sum = 0
  i = 1
  while sequence[i] < val do
    sequence.append(sequence[i]+sequence[i-1])
    if sequence[i].even?
      sum = sum + sequence[i]
    end
    i+=1
  end
  puts sum
end

sum_of_even_fibonachi(4000000)
